test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/LanguageServer/PeekDefinitionTests.cs

```diff
@@ -698,113 +698,122 @@ public void GetDatabaseWithQueryConnectionTest()
-        //     scriptFile.Contents = queryString;
-
-        //     var service = new LanguageService();
-        //     service.BindingQueue = bindingQueue;
-        //     await service.UpdateLanguageServiceOnConnection(connectionResult.ConnectionInfo);
-        //     Thread.Sleep(2000);
-
-        //     ScriptParseInfo scriptInfo = new ScriptParseInfo { IsConnected = true };
-        //     scriptInfo.ConnectionKey = bindingQueue.AddConnectionContext(connInfo);
-        //     service.ScriptParseInfoMap.Add(TestUri, scriptInfo);
-
-        //     // When I call the language service
-        //     var fnResult = service.GetDefinition(fnDocument, scriptFile, connInfo);
-        //     var sysResult = service.GetDefinition(dboDocument, scriptFile, connInfo);
-        //     var masterResult = service.GetDefinition(masterDocument, scriptFile, connInfo);
-
-        //     // Then I expect the results to be non-null
-        //     Assert.NotNull(fnResult);
-        //     Assert.NotNull(sysResult);
-        //     Assert.NotNull(masterResult);
-
-        //     // And I expect the all results to be the same
-        //     Assert.True(CompareLocations(fnResult.Locations, sysResult.Locations));
-        //     Assert.True(CompareLocations(fnResult.Locations, masterResult.Locations));
-
-        //     Cleanup(fnResult.Locations);
-        //     Cleanup(sysResult.Locations);
-        //     Cleanup(masterResult.Locations);
-        //     service.ScriptParseInfoMap.Remove(TestUri);
-        //     connInfo.RemoveAllConnections();
-        // }
+        [Fact]
+        public async void GetDefinitionFromChildrenAndParents()
+        {
+            string queryString = "select * from master.sys.objects";
+
+            // place the cursor on every token
+
+
+            //cursor on objects
+            TextDocumentPosition objectDocument = CreateTextDocPositionWithCursor(26, OwnerUri);
+
+            //cursor on sys
+            TextDocumentPosition sysDocument = CreateTextDocPositionWithCursor(22, OwnerUri);
+
+            //cursor on master
+            TextDocumentPosition masterDocument = CreateTextDocPositionWithCursor(17, OwnerUri);
+
+            LiveConnectionHelper.TestConnectionResult connectionResult = LiveConnectionHelper.InitLiveConnectionInfo(null, OwnerUri);
+            ScriptFile scriptFile = connectionResult.ScriptFile;
+            ConnectionInfo connInfo = connectionResult.ConnectionInfo;
+            connInfo.RemoveAllConnections();
+            var bindingQueue = new ConnectedBindingQueue();
+            bindingQueue.AddConnectionContext(connInfo);
+            scriptFile.Contents = queryString;
+
+            var service = new LanguageService();
+            service.RemoveScriptParseInfo(OwnerUri);
+            service.BindingQueue = bindingQueue;
+            await service.UpdateLanguageServiceOnConnection(connectionResult.ConnectionInfo);
+            Thread.Sleep(2000);

```

Comment:
I followed some other tests that did the same thing. I assumed this is the way to go. The failures were caused by reusing the same ownerURI for every test that uses live connection. I think the new changes should fix that problem.